source/uwp/Renderer/lib/AdaptiveInputs.cpp

```diff
@@ -0,0 +1,48 @@
+#include "pch.h"
+#include "AdaptiveInputs.h"
+
+#include "AdaptiveCard.h"
+#include "AsyncOperations.h"
+#include <windows.foundation.collections.h>
+#include <Windows.UI.Xaml.h>
+#include "XamlHelpers.h"
+
+using namespace concurrency;
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace ABI::AdaptiveCards::XamlCardRenderer;
+using namespace ABI::Windows::Foundation;
+using namespace ABI::Windows::Foundation::Collections;
+using namespace ABI::Windows::Data::Json;
+using namespace ABI::Windows::UI;
+using namespace ABI::Windows::UI::Xaml;
+using namespace ABI::Windows::UI::Xaml::Controls;
+
+namespace AdaptiveCards { namespace XamlCardRenderer
+{
+    AdaptiveInputs::AdaptiveInputs()
```

Comment:
>AdaptiveInputs [](start = 4, length = 14)

I wasn't sure if we wanted to reuse the InputItem class, since that class represents just one input, but the AdaptiveInputs interface represents the collection of all inputs in the card.

Reply:
This approach makes sense to me.

---
In reply to: [140721245](https://github.com/Microsoft/AdaptiveCards/pull/678#discussion_r140721245) [](ancestors = 140721245)